### ISSUE TITLE
Various literals allowed in postfix position on expr

### ISSUE THREAD

**Issue by bishabosha** [lampepfl/dotty#5824]
tested in Dotty REPL 0.12-RC1, I believe only these cases are an issue:
```scala
val foo = <expr> <booleanLiteral
                  | characterLiteral
                  | stringLiteral
                  | symbolLiteral
                  | "null">
```
e.g.
```scala
val foo = 123 'bar;
```

in Dotty REPL results in `val foo: Int = 123`, this is particularly bad as the Symbol literal is ignored, where otherwise an error is generated for their presence.

However, in Scala REPL 2.12.8 results in:
```
<console>:1: error: ';' expected but symbol literal found.
       val foo = 123 'bar;
                     ^
```

**Comment by odersky** [comment]
Seems to be REPL only. I could not reproduce standalone.

**Comment by bishabosha** [comment]
This compiles with dotc.

```scala
object Main {
  def main(args: Array[String]): Unit = {
    val foo = 123 "";
    println(foo)
  }
}
```

**Comment by Blaisorblade** [comment]
Tokens are being swallowed and ignored — seems a dup of #4934.

**Comment by odersky** [comment]
Fixed by #6209
